src/VisualStudio/Core/Def/Venus/ContainedDocument.cs

```diff
@@ -184,14 +185,33 @@ public DocumentId FindProjectDocumentIdWithItemId(uint itemidInsertionPoint)
                 }
             }
 
+            // The requested insertion itemid wasn't found. Check if the containing
+            //   document matches the requested itemid.
+            var containingDocumentItemid = GetContainingDocumentItemId(hierarchy);
+            if (containingDocumentItemid == itemidInsertionPoint)
+            {
+                var containingDocument = GetOpenTextContainer().CurrentText.GetOpenDocumentInCurrentContextWithChanges();
+
+                return containingDocument.Id;
+            }
+
             return null;
         }
 
         public uint FindItemIdOfDocument(Document document)
         {
             // We cast to VisualStudioWorkspace because the expectation is this isn't being used in Live Share workspaces
             var hierarchy = ((VisualStudioWorkspace)_workspace).GetHierarchy(_project.Id);
-            return hierarchy.TryGetItemId(_workspace.CurrentSolution.GetDocument(document.Id).FilePath);
+            var itemId = hierarchy.TryGetItemId(_workspace.CurrentSolution.GetDocument(document.Id).FilePath);
+
+            if (itemId == (uint)VSITEMID.Nil)
```

Comment:
```suggestion
            if (document.Id == this.Id)
```

Otherwise Document could be anything else random and we'd return the wrong thing, right? Can we then put this check before the call out to the hierarchy, or did you want to leave the existing path in first?

Reply:
Yeah, missed the existing Id property on this class. Good catch!